src/pkzip.c

```diff
@@ -248,3 +266,10 @@ void *winzip_common_binary(char *ciphertext) {
 	}
 	return bin;
 }
+
+unsigned int winzip_common_cost_hmac_len(void *salt)
+{
+	winzip_salt *s = *((winzip_salt**)salt);
+
+	return (s->comp_len + 1023) / 1024;
```

Comment:
This will silently overflow above 4 TiB. Perhaps pre-check for that in `valid` and refuse to load, for 64-bit hosts with 8+ TiB of RAM.

Reply:
> This will silently overflow above 4 TiB. Perhaps pre-check for that in `valid` and refuse to load, for 64-bit hosts with 8+ TiB of RAM.

So 8+ TB of input... Even if that could theoretically reach this code, I think this is overkill. And it would take so very long time to get here that any user would cancel the run.